File: src/main/java/arc/arc/hooks/viaversion/ViaVersionHook.java
package arc.arc.hooks.viaversion;

import com.viaversion.viaversion.api.Via;
import com.viaversion.viaversion.api.connection.UserConnection;
import org.bukkit.entity.Player;

public class ViaVersionHook {

    public int getPlayerVersion(Player player) {
        UserConnection userConnection = Via.getManager().getConnectionManager().getConnectedClient(player.getUniqueId());
        if (userConnection != null) {
            System.out.println("Version: "+userConnection.getProtocolInfo().getProtocolVersion());
            return userConnection.getProtocolInfo().getProtocolVersion();
        }
        return -1; // Return -1 if unable to get the version
    }

}

File: src/main/kotlin/ru/arc/hooks/viaversion/ViaVersionHook.kt
package ru.arc.hooks.viaversion

import com.viaversion.viaversion.api.Via
import org.bukkit.entity.Player

class ViaVersionHook {
    fun getPlayerVersion(player: Player): Int {
        val connection = Via.getManager().connectionManager.getConnectedClient(player.uniqueId)
        return connection?.protocolInfo?.protocolVersion ?: -1
    }
}
